### ISSUE TITLE
Kronecker solver

### ISSUE THREAD

**Issue by iosonobert** [TIDE-ITRH/gptide#13]
Need a class for handling Kronecker inversions. Proposed method:

- New class GPKronecker
- Keep init function from the base class
- gpkronecker.GPKronecker._calc_cov returns a list of covariance matrices which combine by Kronecker product to produce K
- gpkronecker.GPKronecker._calc_weights returns a list of choleskis of the same size
- v = la.cho_solve((self.L, True),  Kdm) line is replaces with series of la.cho_solve((self.L, True),  np.np.eye(self.M)) to give inverses, which are kroneckered together to give sigma inv, then np.matmul with Kdm

**Comment by iosonobert** [comment]
Kronecker+Toeplitz would be next

**Comment by mrayson** [comment]
Is this for high dimensional and/or multivariable models?

**Comment by astfalckl** [comment]
Yeah. Anything we can Kronecker the covariance with. It should, in theory, dial back the estimation time on bivariate things like Larry by about a factor of 8.

**Comment by iosonobert** [comment]
Yep. So the first test case would be to repeat Larry with this solver. But instead of Kronecker inside covfunc return the two matrices and kronecker after inversion. Should go from about an hour to ~8 minutes.   

Ultimately I'll need this to do PoSO U+V together. I'm at 30 mins univariate for a 15 day record subsampled to hourly data.

**Comment by mrayson** [comment]
One idea for all classes would be to introduce a `solve` and `calc_logdet` method. That way all the trickery required for this case could be buried in there

**Comment by iosonobert** [comment]
Yep. And another way to unify trickery is that if cov is just a single cov matrix make the list cov = [np.array([1]), cov] and always use kronecker

**Comment by astfalckl** [comment]
`a1 = cho_solve(Lt, Y)`
`a2 = cho_solve(Lo, solve1)`
`alpha = vec(a2^T)`
